### PR TITLE
User specific insights

### PR DESCRIPTION
**Pull request by willstenzel**
This PR allows a user to see only the insights that they care about. Insights can now be toggled on/off from the insights settings page or hidden by directly clicking the hide icon on an insight card in the dashboard.

### CODE REVIEW

File: packages/app/components/Settings/SettingsPage.tsx [hunk 1]
```diff
@@ -116,6 +128,14 @@ export default function SettingsPage({
           >
             Notifications Settings
           </Menu.Item>
+          {role === Role.PROFESSOR ? (
```
Comment (NEUDitao) [inline packages/app/components/Settings/SettingsPage.tsx:131]: this could just be an `&&` right?

File: packages/app/pages/dev.tsx [hunk 2]
```diff
@@ -53,28 +57,28 @@ export default function DevPanel(): ReactElement {
           <h3>Login</h3>
         </Divider>
         <a href="/api/v1/login/dev?userId=1">
-          <DevPageButton type="default" htmlType="submit">
+          <Button style={{ margin: "15px" }} type="default" htmlType="submit">
```
Comment (NEUDitao) [inline packages/app/pages/dev.tsx:60]: why're we unstyling this?

File: packages/app/components/Settings/InsightsSettings.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import { useProfile } from "../../hooks/useProfile";
+
+export default function InsightsSettings(): ReactElement {
+  const profile = useProfile();
+  const { data: insightsList, error, mutate } = useSWR(
+    `api/v1/insights`,
+    async () => API.insights.list()
+  );
+
+  return (
```
[the remark this answers is not in the document]
Comment (willstenzel) [reply packages/app/components/Settings/InsightsSettings.tsx:14]: Either way works, I don't have a preference either way

File: packages/app/components/Settings/SettingsPage.tsx (hunk 1 above)
Comment (willstenzel) [reply packages/app/components/Settings/SettingsPage.tsx]: yep, got that changed

File: packages/app/pages/dev.tsx (hunk 2 above)
Comment (willstenzel) [reply packages/app/pages/dev.tsx]: Whoops.. the old version must have slipped in when I merged. Got that fixed up

File: packages/app/components/Settings/InsightsSettings.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (willstenzel) [reply packages/app/components/Settings/InsightsSettings.tsx]: @NEUDitao after thinking about it, you're right. The "Insights Settings" page is better off on the dashboard itself. I made the changes in this commit https://github.com/sandboxnu/office-hours/pull/503/commits/dd007ac86b93fdd39101a652ddbe6b9e56aaa9fe so if you could re-review that would be great.

File: packages/app/pages/course/[cid]/insights.tsx [hunk 4]
```diff
@@ -1,40 +1,84 @@
...
 import SimpleChartComponent from "../../../components/Insights/components/SimpleChartComponent";
 import SimpleDisplayComponent from "../../../components/Insights/components/SimpleDisplayComponent";
+import InsightsDisplayOptions from "../../../components/Insights/components/InsightsDisplayOptions";
 
 export default function Insights(): ReactElement {
-  // TODO: In the future this will come from the users specific insights that want to see
-  const insights = ["TotalQuestionsAsked", "TotalStudents"];
+  const router = useRouter();
+  const { cid } = router.query;
+  const { data: profile, error, mutate } = useSWR(`api/v1/profile`, async () =>
```
Comment (NEUDitao) [inline packages/app/pages/course/[cid]/insights.tsx:18]: food for thought, but not this pr, should we move `useProfile` to also export `data, error`, and `mutate`? It seemsl ike we need it a lot these days

File: packages/server/src/insights/course-role.decorator.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import { createParamDecorator, ExecutionContext } from '@nestjs/common';
+import { UserModel } from 'profile/user.entity';
+
+export const CourseRole = createParamDecorator(
```
Comment (NEUDitao) [inline packages/server/src/insights/course-role.decorator.ts:4]: yo wait this is lit

**Review by NEUDitao — APPROVED** [review]

File: packages/app/pages/course/[cid]/insights.tsx (hunk 4 above)
Comment (willstenzel) [reply packages/app/pages/course/[cid]/insights.tsx:18]: Yeah I totally agree.

**Review by dajinchu — APPROVED** [review]

**Pull request merged**